config/application.rb

```diff
@@ -86,5 +86,8 @@ class Application < Rails::Application
     config.gtm_id = ENV.fetch("GTM_ID", nil)
     config.tidelift_api_key = ENV.fetch("TIDELIFT_API_KEY", nil)
     config.pg_search_projects_enabled = ENV.key?("PG_SEARCH_PROJECTS_ENABLED")
+
+    # In rails 7.1 we can use autoload_lib instead
```

Comment:
wait this is just making one method reusable across a bunch of classes why do we need to make it an extension vs just a utils method...?

Reply:
I prefer extensions defined on the primitive types, a la ActiveSupport::CoreExtensions, instead of util classes, and it also seems like a common Ruby convention and I never saw a downside to it.

Reply:
The method being added is really only applicable to a specialized type of hashes, so I'm not sure if it's really a globally useful method for all hashes.

Some of the downsides I can see are that it makes it harder to find where this method comes from if it needs to be debugged, as it's a method that's custom but not in an easily accessible class that most custom util methods are expected to be in. If all utils get loaded in this way and there's a name collision in a different utility package then it'll be harder to find and rectify, especially if its an edge case or if another library takes up the name. I'm not sure what this means for application start/load order but it might become a performance drag on application loading time.

Reply:
> The method being added is really only applicable to a specialized type of hashes, so I'm not sure if it's really a globally useful method for all hashes.

the method I'm adding here is a string extension: `String#strip_null_bytes`

> Some of the downsides I can see are that it makes it harder to find where this method comes from if it needs to be debugged

`"some string".method(:strip_null_bytes).source_location`

Reply:
I've been fine with the CoreExtensions pattern for the past 19 years I've been using it, but if y'all are more comfortable with a utils class I can do that.

Reply:
I don't want to belabor my point and I've already pushed the "StringUtils" class instead, but it seems like a good moment to point out how common it is to extend primitives in Ruby:

```
 (String.instance_methods - Object.methods).map { |s| [s, String.instance_method(s).source_location] }.reject { |a| a[1].nil? || a[1][0] =~ /\/ruby\/3\.2\.0|internal/ }.sort_by(&:first)

[:acts_like_string?, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/behavior.rb", 5]],
 [:at, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/access.rb", 29]],
 [:camelcase, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 101]],
 [:camelize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 101]],
 [:classify, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 239]],
 [:column_of, ["lib/ruby/gems/3.2.0/gems/treetop-1.6.12/lib/treetop/ruby_extensions/string.rb", 2]],
 [:constantize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 73]],
 [:dasherize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 148]],
 [:deconstantize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 177]],
 [:demodulize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 162]],
 [:downcase_first, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 284]],
 [:exclude?, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/exclude.rb", 10]],
 [:ext, ["lib/ruby/gems/3.2.0/gems/rake-13.1.0/lib/rake/ext/string.rb", 14]],
 [:first, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/access.rb", 78]],
 [:foreign_key, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 297]],
 [:from, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/access.rb", 46]],
 [:html_safe, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/output_safety.rb", 232]],
 [:humanize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 262]],
 [:in_time_zone, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/zones.rb", 9]],
 [:indent, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/indent.rb", 42]],
 [:indent!, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/indent.rb", 7]],
 [:inquiry, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inquiry.rb", 13]],
 [:is_utf8?, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/multibyte.rb", 48]],
 [:last, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/access.rb", 92]],
 [:line_of, ["lib/ruby/gems/3.2.0/gems/treetop-1.6.12/lib/treetop/ruby_extensions/string.rb", 12]],
 [:mb_chars, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/multibyte.rb", 37]],
 [:parameterize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 215]],
 [:pathmap, ["lib/ruby/gems/3.2.0/gems/rake-13.1.0/lib/rake/ext/string.rb", 138]],
 [:pathmap_explode, ["lib/ruby/gems/3.2.0/gems/rake-13.1.0/lib/rake/ext/string.rb", 27]],
 [:pathmap_partial, ["lib/ruby/gems/3.2.0/gems/rake-13.1.0/lib/rake/ext/string.rb", 41]],
 [:pathmap_replace, ["lib/ruby/gems/3.2.0/gems/rake-13.1.0/lib/rake/ext/string.rb", 59]],
 [:pluralize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 35]],
 [:remove, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/filters.rb", 32]],
 [:remove!, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/filters.rb", 40]],
 [:safe_constantize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 86]],
 [:singularize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 60]],
 [:squish, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/filters.rb", 13]],
 [:squish!, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/filters.rb", 21]],
 [:strip_heredoc, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/strip.rb", 22]],
 [:tableize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 227]],
 [:tabto, ["lib/ruby/gems/3.2.0/gems/treetop-1.6.12/lib/treetop/ruby_extensions/string.rb", 17]],
 [:titlecase, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 126]],
 [:titleize, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 126]],
 [:to, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/access.rb", 63]],
 [:to_bool, ["/Users/tiegzaharia/Sites/dependencyci/lib/extensions/string/to_bool.rb", 4]],
 [:to_d, ["lib/ruby/gems/3.2.0/gems/bigdecimal-3.1.6/lib/bigdecimal/util.rb", 72]],
 [:to_date, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/conversions.rb", 47]],
 [:to_datetime, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/conversions.rb", 57]],
 [:to_json_raw, ["lib/ruby/gems/3.2.0/gems/oj-3.13.21/lib/oj/mimic.rb", 289]],
 [:to_json_raw_object, ["lib/ruby/gems/3.2.0/gems/oj-3.13.21/lib/oj/mimic.rb", 283]],
 [:to_msgpack, ["lib/ruby/gems/3.2.0/gems/msgpack-1.7.2/lib/msgpack/core_ext.rb", 3]],
 [:to_time, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/conversions.rb", 22]],
 [:treetop_camelize, ["lib/ruby/gems/3.2.0/gems/treetop-1.6.12/lib/treetop/ruby_extensions/string.rb", 31]],
 [:truncate, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/filters.rb", 70]],
 [:truncate_bytes, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/filters.rb", 101]],
 [:truncate_words, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/filters.rb", 142]],
 [:underscore, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 139]],
 [:upcase_first, ["lib/ruby/gems/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/string/inflections.rb", 273]]]
```